### PR TITLE
molsketch: init at 0.7.3

### PR DESCRIPTION
**Pull request by moni-dz**
###### Description of changes
2d molecule editor

image

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/molsketch/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, mkDerivation, fetchgit, cmake, pkg-config, hicolor-icon-theme, openbabel, desktop-file-utils, qttranslations }:
```
Comment (markuskowa) [inline pkgs/applications/editors/molsketch/default.nix:1]: This is a long line.
```suggestion
{ lib
, stdenv
, mkDerivation
, fetchgit
, cmake
, pkg-config
, hicolor-icon-theme
, openbabel
, desktop-file-utils
, qttranslations 
}:
```

File: pkgs/applications/editors/molsketch/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    rev = "61c019eef03b978247c2a8a0c38c0578ce8d3aee";
+    hash = "sha256-VJPCaMOo2x4o4FHZy4wHR1AA7nfXYwzti/c7YZcrCr4=";
+  };
+
+  preConfigure = ''
+    cmakeFlags="$cmakeFlags -DMSK_PREFIX=$out"
+  '';
+
+  nativeBuildInputs = [ cmake pkg-config ];
+  buildInputs = [ hicolor-icon-theme openbabel desktop-file-utils qttranslations ];
```
Comment (markuskowa) [inline pkgs/applications/editors/molsketch/default.nix:18]:
```suggestion
  buildInputs = [ 
    hicolor-icon-theme 
    openbabel 
    desktop-file-utils 
    qttranslations 
  ];
```

File: pkgs/applications/editors/molsketch/default.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, mkDerivation, fetchgit, cmake, pkg-config, hicolor-icon-theme, openbabel, desktop-file-utils, qttranslations }:
+
+mkDerivation {
+  pname = "molsketch";
+  version = "unstable-2022-09-18";
```
Comment (markuskowa) [inline pkgs/applications/editors/molsketch/default.nix:5]: Is there a particular reason, why you want to use an unstable version? The latest stable is just 3 months old.

**Review by markuskowa — CHANGES_REQUESTED** [review]

File: pkgs/applications/editors/molsketch/default.nix (hunk 3 above)
Comment (fortuneteller2k) [reply pkgs/applications/editors/molsketch/default.nix:15]: I believe this is the same revision as the latest stable (?).  I didn't want to use the tarballs and instead directly get the source from git, but upstream didn't have a tag for the stable release.

Comment (markuskowa) [reply pkgs/applications/editors/molsketch/default.nix:15]: There's nothing wrong with tar balls for that purpose. I would recommend here to pick the stable version from sourceforge with a mirror URL (i.e. ` url = "mirror://sourceforge/...`).

**Review by markuskowa — APPROVED** [review]

**Pull request merged**